lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +493,1338 @@ LocaleMatch lookupMatcher(
   return x.compare(y);
 }
 
+// Implementation of
+// https://tc39.es/ecma402/#datetimeformat-objects
 struct DateTimeFormat::Impl {
-  std::u16string locale;
+  struct newRecord {
+    std::u16string localeMatcher, ca, nu, hc, hcDefault; // For opt
+  };
+  // Table 7
+  // https://tc39.es/ecma402/#table-datetimeformat-resolvedoptions-properties
+  // For DateTimeFormat::impl_ in initialize
+  std::u16string FormatMatcher, HourCycle, Weekday, Era, Year, Month, Day,
+      DayPeriod, Hour, Minute, Second, TimeZone, TimeZoneName, DateStyle,
+      TimeStyle, Calendar, NumberingSystem, locale;
+  uint8_t FractionalSecondDigits;
+  bool hour12;
 };
 
 DateTimeFormat::DateTimeFormat() : impl_(std::make_unique<Impl>()) {}
 DateTimeFormat::~DateTimeFormat() {}
 
+// Implementation of
+// https://tc39.es/ecma402/#sec-resolvelocale
+struct ResolveLocale {
+  std::u16string key, value, dataLocale, resolveLocale, locale, ca, nu, hc;
+};
+struct UnicodeExtensionKeys {
+  std::u16string CALENDAR = u"calendar";
+  std::u16string CALENDAR_CANON = u"ca";
+
+  std::u16string NUMERINGSYSTEM = u"numbers";
+  std::u16string NUMERINGSYSTEM_CANON = u"nu";
+
+  std::u16string HOURCYCLE = u"hours";
+  std::u16string HOURCYCLE_CANON = u"hc";
+
+  std::u16string COLLATION = u"collation";
+  std::u16string COLLATION_CANON = u"co";
+
+  std::u16string COLLATION_NUMERIC = u"colnumeric";
+  std::u16string COLLATION_NUMERIC_CANON = u"kn";
+
+  std::u16string COLLATION_CASEFIRST =
+      u"colcasefirst"; // TODO:: double check this
+  std::u16string COLLATION_CASEFIRST_CANON = u"kf";
+};
+std::u16string resolveCalendarAlias(const std::u16string &value) {
+  // https://github.com/unicode-org/cldr/blob/master/common/bcp47/calendar.xml
+  static std::vector<std::u16string> s_calendarAliasMappings = {
+      u"gregorian", u"gregory"};
+  if (std::find(
+          s_calendarAliasMappings.begin(),
+          s_calendarAliasMappings.end(),
+          value) == s_calendarAliasMappings.end()) {
+    return value;
+  } else {
+    int pos = find(
```

Comment:
1. What is the difference in behaviour between `find` and `std::find`?
2. What happens if value is not in s_calendarAliasMappings? Will app crash trying to access `s_calendarAliasMappings.end()`?
3. I suggest extracting lookup procedure into a separate function since it is the same for all three `resolve*Alias` functions.

Reply:
1. Sorry, that looks like a typo they should both be the same. 
2. I tried to emulate https://github.com/facebook/hermes/blob/c65a5ebb12621ae866501051923cbe397dfaa3f8/lib/Platform/Intl/java/com/facebook/hermes/intl/UnicodeExtensionKeys.java#L110 , but I’m not entirely sure what these AliasMappings functions are trying to achieve as it just returns value, and yes it does feel odd that there isn’t any error checking in the function.

Reply:
Thank you sharing this link.
`s_calendarAliasMappings`  in this Java implementation is an actual map. Its type is `Map<String, String>`. This means that it represents mapping from `u"gregorian"` to `u"gregory"`. `u"gregorian"` is the key, and `u"gregory"` is the value.
Their lookup procedure checks if there is a key in the map, if yes - it returns corresponding value, if not - it returns the key itself.
In order to be able to implement this functionality in C++, you should be using `std::map` instead of `std::vector`.

Reply:
It is also true for `resolveNumberSystemAlias` and `resolveCollationAlias`.

Reply:
Ah right thank you that makes sense now. So no error checking is needed as when the value isn’t found in the map, it will just return the value.